### ISSUE TITLE
Code for Mini-Kaggle

### ISSUE THREAD

**Issue by ArmandGiraud** [ArmandGiraud/Projet-Kaggle#2]
# Projet-Kaggle
### Call the following packages (to be run in Jupyter)

```
%matplotlib inline
import numpy as np
import matplotlib.pyplot as plt
np.random.seed(1) 
```

# Génère les séries x et y

```
x = np.linspace(-3, 3, num = 50) # x est définit comme une séquence continue de 50 nombres de -3 à 3
y = np.exp(-x**2) + 0.1 * np.random.randn(50) # ...

x = x.reshape(-1,1) # permet de redéfinir les variables comme des Arrays (format lisible par numpy)
y = y.reshape(-1,1)


plt.scatter(x,y,color='r') #On visualise les données de base
plt.xlabel('X')
plt.ylabel('Y')
plt.title('training set')
```

# Générer un test set avec 10 observations pour évaluer le modèle

```
testx = np.random.uniform(3,-3,size = 10)
testy = np.exp(-testx**2)
testx = testx.reshape(-1,1)
testy = testy.reshape(-1,1)
plt.scatter(testx,testy)

```
![Plot avec training set en rouge et test set en bleu](https://cloud.githubusercontent.com/assets/22575341/19627781/8352e8fc-994f-11e6-92c8-fc1dd7018fc7.png)
##Exemple régression linéaire

```
from sklearn import linear_model
lm = linear_model.LinearRegression()
lm.fit(x,y) # linear model
plt.scatter(x,y)
plt.plot(x,lm.predict(x),color="r",linewidth=3)
#Rappel calculer la MAE

print("Mean Absolute Error: %.2f"
    % np.mean(abs(lm.predict(testx) - testy)))
```
![plot2](https://cloud.githubusercontent.com/assets/22575341/19627799/fa1a6190-994f-11e6-9743-b5acea9bf596.png)
